File: sample_vulnerable.cpp
// Sample vulnerable C++ code for testing Sentinel-AI
// This file contains intentional security vulnerabilities

#include <iostream>
#include <string>
#include <fstream>
#include <cstdlib>
#include <mysql/mysql.h>

using namespace std;

// Hardcoded credentials (security risk)
const char* DB_HOST = "localhost";
const char* DB_USER = "root";
const char* DB_PASS = "password123";
const char* DB_NAME = "mydb";

int main(int argc, char* argv[]) {
    if (argc < 2) {
        cout << "Usage: " << argv[0] << " <user_id>" << endl;
        return 1;
    }

    string userId = argv[1];
    MYSQL* conn;
    MYSQL_RES* res;
    MYSQL_ROW row;

    // Initialize MySQL connection
    conn = mysql_init(NULL);

    if (!mysql_real_connect(conn, DB_HOST, DB_USER, DB_PASS, DB_NAME, 0, NULL, 0)) {
        cerr << "Connection failed: " << mysql_error(conn) << endl;
        return 1;
    }

    // SQL Injection vulnerability: Direct string concatenation
    string query = "SELECT * FROM users WHERE id = " + userId;

    if (mysql_query(conn, query.c_str())) {
        cerr << "Query failed: " << mysql_error(conn) << endl;
        mysql_close(conn);
        return 1;
    }

    res = mysql_use_result(conn);

    // Process results
    while ((row = mysql_fetch_row(res)) != NULL) {
        cout << "User: " << row[0] << endl;
    }

    mysql_free_result(res);
    mysql_close(conn);

    return 0;
}

// Command injection vulnerability
void pingHost(const string& host) {
    // Vulnerable: User input passed directly to system command
    string command = "ping -c 4 " + host;
    system(command.c_str());
}

// Path traversal vulnerability
void readFile(const string& filename) {
    // Vulnerable: User input used directly in file path
    string filepath = "/var/www/files/" + filename;

    ifstream file(filepath.c_str());
    if (file.is_open()) {
        string line;
        while (getline(file, line)) {
            cout << line << endl;
        }
        file.close();
    }
}

File: sample_vulnerable.js
// Sample vulnerable JavaScript code for testing Sentinel-AI
// This file contains intentional security vulnerabilities

const express = require('express');
const mysql = require('mysql');
const app = express();

// SQL Injection vulnerability
app.get('/user/:id', (req, res) => {
    const userId = req.params.id;
    // Vulnerable: Direct string concatenation in SQL query
    const query = "SELECT * FROM users WHERE id = " + userId;

    const connection = mysql.createConnection({
        host: 'localhost',
        user: 'root',
        password: 'password123', // Hardcoded credentials
        database: 'mydb'
    });

    connection.query(query, (error, results) => {
        if (error) throw error;
        res.json(results);
    });
});

// XSS vulnerability
app.get('/welcome', (req, res) => {
    const name = req.query.name;
    // Vulnerable: User input directly inserted into HTML
    const html = `<h1>Welcome ${name}!</h1>`;
    res.send(html);
});

// Command injection vulnerability
app.get('/ping/:host', (req, res) => {
    const host = req.params.host;
    const { exec } = require('child_process');
    // Vulnerable: User input passed to system command
    exec(`ping ${host}`, (error, stdout, stderr) => {
        res.send(stdout);
    });
});

app.listen(3000, () => {
    console.log('Server running on port 3000');
});